### PR TITLE
Make V5 default Role version

### PR DESCRIPTION
**Pull request by Joerger**
This PR updates the default role version to V5 rather than V3. This way, consumers of the Teleport API, such as the [Teleport Terraform Plugin](https://github.com/gravitational/teleport-plugins/tree/master/terraform), will create new roles as V5 roles rather than V3 roles. Existing V3 roles and V4 roles will not be changed.

Internal roles (presets, test roles, etc) remain unchanged, since they now use `NewRoleV3`.

### CODE REVIEW

**Comment by Joerger** [comment]
@xacrimon @nklaassen What is our long term plan for deprecating V3 and V4 roles? Do we just need to update our preset roles and other roles used in tests to V5?

File: api/types/role.go [hunk 1]
```diff
@@ -159,11 +159,10 @@ type Role interface {
 	GetSessionPolicySet() SessionTrackerPolicySet
 }
 
-// NewRole constructs new standard V3 role.
-// This is mostly a legacy function and will create a role with V3 RBAC semantics.
+// NewRole constructs new standard V5 role.
+// This creates a V5 role with V4+ RBAC semantics. This should be preferred over `NewRole`.
```
Comment (nklaassen) [inline api/types/role.go:163]:
```suggestion
// This creates a V5 role with V4+ RBAC semantics.
```

**Review by nklaassen — COMMENTED** [review]
beware `GetCheckerForBuiltinRole` heavily relies on `services.FromSpec` which calls `types.NewRole`. Will changing the default to v5 here silently change all of the builtin roles and remove the current wildcard defaults?

IIRC the blocker for changing everything to the latest version by default has been the builtins. Last time I tried putting all the required wildcards on the builtin roles explicitly it looked pretty scary and got rejected in review. We keep kicking the can down the road on this one but we should really bite the bullet and fix it sooner than later.

**Comment by xacrimon** [comment]
Yup, as @nklaassen said, I wanted to make the `NewRole` function default to V5 but ended up calling it off since it broke every test imaginable due to label matching and caused all kinds of funkiness on new clusters. Agreed, at some point we need to bite the bullet, this is going to have to either get into v9 or get pushed to v10 as it's a major breaking change depending on how long it takes to test/verify and stabilize everything incl plugins.

The plan so far has been "this is scary, let's try to deprecate it later".

**Comment by xacrimon** [comment]
@stevenGravy Made changes already to change the default to v5 in the web editor for new roles which is a good step forward but builtins and defaults remain at v3.

**Comment by Joerger** [comment]
Makes sense, I suppose the best thing we can do is make the change over a wider span of time and make the changes less intimidating. What do you guys think of starting with this PR as a logical first step to make V5 the default without changing any internals and clarify how V3 roles are still being used?

Then we can make one or a few more PRs to update builtins, tests, etc. And we can test this throughout `10.0.0-dev` to ensure nothing breaks?

> beware `GetCheckerForBuiltinRole` heavily relies on `services.FromSpec` which calls `types.NewRole`. Will changing the default to v5 here silently change all of the builtin roles and remove the current wildcard defaults?

My intention was to make All callers of `types.NewRole` call `types.NewRoleV3`. This is now the case with the latest commit, so builtin roles will be V3 still.

**Review by nklaassen — APPROVED** [review]
This is a good first step imo, LGTM now that `services.FromSpec` has been updated

**Review by timothyb89 — APPROVED** [review]

**Review by r0mant — COMMENTED** [review]
@Joerger With the migration to RoleV5, please verify that downgrades are still possible (i.e. if you upgrade existing v8 cluster to v9 and then downgrade, it will still start and won't override existing roles to v9).

**Pull request merged**